crates/iota-indexer/src/store/pg_indexer_store.rs

```diff
@@ -279,7 +291,80 @@ impl PgIndexerStore {
+            format!("Failed reading transaction range from PostgresDB for checkpoint {checkpoint}")
+                .as_str(),
+        )
+    }
+
+    pub(crate) async fn get_global_order_for_tx_seq_in_blocking_worker(
+        &self,
+        tx_seq: i64,
+    ) -> Result<(i64, i64), IndexerError> {
+        self.execute_in_blocking_worker(move |this| this.get_global_order_for_tx_seq(tx_seq))
+            .await
+    }
+
+    fn get_global_order_for_tx_seq(&self, tx_seq: i64) -> Result<(i64, i64), IndexerError> {
+        read_only_blocking!(&self.blocking_cp, |conn| {
+            tx_global_order::dsl::tx_global_order
+                .select((
+                    tx_global_order::global_sequence_number,
+                    tx_global_order::optimistic_sequence_number,
+                ))
+                .filter(tx_global_order::chk_tx_sequence_number.eq(tx_seq))
+                .first::<(i64, i64)>(conn)
+        })
+        .context(
+            format!("Failed reading global sequence number from PostgresDB for tx seq {tx_seq}")
+                .as_str(),
+        )
+    }
+
+    pub(crate) async fn prune_optimistic_transactions_up_to_in_blocking_worker(
+        &self,
+        to: (i64, i64),
+        limit: i64,
+    ) -> IndexerResult<usize> {
+        self.execute_in_blocking_worker(move |this| {
+            this.prune_optimistic_transactions_up_to(to, limit)
+        })
+        .await
+    }
+
+    fn prune_optimistic_transactions_up_to(
+        &self,
+        to: (i64, i64),
+        limit: i64,
+    ) -> IndexerResult<usize> {
+        let (to_global_sequence_num, to_optimistic_sequence_num) = to;
+        transactional_blocking_with_retry!(
+            &self.blocking_cp,
+            |conn| {
+                let sql = "\
+                    WITH ids_to_delete AS (\
+                         SELECT global_sequence_number, optimistic_sequence_number \
+                         FROM optimistic_transactions \
+                         WHERE (global_sequence_number, optimistic_sequence_number) <= ($1, $2) \
+                         ORDER BY global_sequence_number, optimistic_sequence_number \
+                         FOR UPDATE LIMIT $3 \
+                     ) \
+                     DELETE FROM optimistic_transactions ot \
+                     USING ids_to_delete \
+                     WHERE (ot.global_sequence_number, ot.optimistic_sequence_number) = \
+                           (ids_to_delete.global_sequence_number, ids_to_delete.optimistic_sequence_number)";
+                diesel::sql_query(sql)
+                    .bind::<BigInt, _>(to_global_sequence_num)
+                    .bind::<BigInt, _>(to_optimistic_sequence_num)
+                    .bind::<BigInt, _>(limit)
+                    .execute(conn)
+                    .map_err(IndexerError::from)
+                    .context(
+                        format!("Failed to prune optimistic_transactions table to {to:?} with limit {limit}").as_str(),
```

Comment:
lowercase for erro messages 

```suggestion
                        format!("failed to prune optimistic_transactions table to {to:?} with limit {limit}").as_str(),
```